src/runtime-mpi.c

```diff
@@ -0,0 +1,188 @@
+
+#include "config.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <mpi.h>
+
+
+#include "runtime.h"
+#include "shmem_internal.h"
+#include "shmem_env.h"
+#include "uthash.h"
+
+#define MAX_KV_COUNT 20
+#define MAX_KV_LENGTH 64
+
+static int rank = -1;
+static int size = 0;
+static MPI_Comm SHMEM_RUNTIME_WORLD;
+static int length = 0;
```

Comment:
Can we rename `length` to something like `kv_length` or `kv_count` since it's global state for the KVS?